oarepo_requests/ui/theme/assets/semantic-ui/js/oarepo_requests_ui/record-requests/components/CreateRequestButtonGroup.jsx

```diff
@@ -1,63 +1,145 @@
 import React from "react";
-
 import { i18next } from "@translations/oarepo_requests_ui/i18next";
-import { Segment, Header, Button, Placeholder, Message, Icon } from "semantic-ui-react";
+import { Button, Placeholder, Message, Confirm } from "semantic-ui-react";
 import _isEmpty from "lodash/isEmpty";
-
 import { RequestModal, CreateRequestModalContent } from ".";
-import { mapLinksToActions } from "./actions";
-import { useRequestContext } from "../contexts";
-
-/**
- * @typedef {import("../types").Request} Request
- * @typedef {import("../types").RequestType} RequestType
- */
+import {
+  useRequestContext,
+  CreateSubmitAction,
+  ConfirmModalContextProvider,
+} from "@js/oarepo_requests_common";
+import PropTypes from "prop-types";
+import { useIsMutating } from "@tanstack/react-query";
 
 /**
- * @param {{ requestTypes: RequestType[], isLoading: boolean, loadingError: Error }} props
+ * @param {{  applicableRequestsLoading: boolean, applicableRequestsLoadingError: Error }} props
  */
-export const CreateRequestButtonGroup = ({ recordLoading, recordLoadingError }) => {
-  const { requestTypes } = useRequestContext();
-  const createRequests = requestTypes.filter(requestType => requestType.links.actions?.create);
-
-  return (
-    <Segment className="requests-create-request-buttons borderless">
-      <Header size="small" className="detail-sidebar-header">{i18next.t("Requests")}</Header>
-      {recordLoading ?
+export const CreateRequestButtonGroup = ({
+  applicableRequestsLoading,
+  applicableRequestsLoadingError,
+}) => {
+  const { requestTypes, requestButtonsIconsConfig } = useRequestContext();
+  const createRequests = requestTypes?.filter(
+    (requestType) => requestType.links.actions?.create
+  );
+  const isMutating = useIsMutating();
+  if (applicableRequestsLoading) {
+    return (
+      <div className="requests-create-request-buttons borderless">
```

Comment:
This wrapper should be lifted above conditions (it is rendered & duplicated for all possible cases)